zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -1202,8 +1205,26 @@ private static Either<List<String>, SearchQueryPage> toSearchQueryPage(
       return Either.right(null);
     }
 
-    final Object[] searchAfter = toArrayOrNull(requestedPage.getSearchAfter());
-    final Object[] searchBefore = toArrayOrNull(requestedPage.getSearchBefore());
+    final List<Object> searchBeforeValues =
+        ofNullable(requestedPage.getSearchBefore())
+            .map(
+                pageObjects ->
+                    pageObjects.stream()
+                        .map(SearchQueryRequestMapper::deserializePageObject)
+                        .collect(Collectors.toList()))
+            .orElse(emptyList());
+
+    final List<Object> searchAfterValues =
+        ofNullable(requestedPage.getSearchAfter())
+            .map(
+                pageObjects ->
+                    pageObjects.stream()
+                        .map(SearchQueryRequestMapper::deserializePageObject)
+                        .collect(Collectors.toList()))
+            .orElse(emptyList());
+
+    final Object[] searchAfter = toArrayOrNull(searchAfterValues);
```

Comment:
❓ Can we directly create an array from the stream instead of creating a list there and transforming it into an array afterward?